xbmc/pvr/recordings/PVRRecording.cpp

```diff
@@ -145,6 +145,10 @@ CPVRRecording::CPVRRecording(const PVR_RECORDING& recording, unsigned int iClien
       }
     }
   }
+  if (recording.mediaType == PVR_RECORDING_MEDIA_TYPE_RECORDING)
+    m_mediaType = RecordingMediaType::PVR_RECORDING_MEDIA_TYPE_RECORDING;
+  else
+    m_mediaType = RecordingMediaType::PVR_RECORDING_MEDIA_TYPE_VOD;
```

Comment:
To be safe against missing future extensions of the add-on API enum here, I would check all known values and add a final else with an error/warning log that we're facing an unknown/unhandled value here.

```c++
else if (recording.mediaType == PVR_RECORDING_MEDIA_TYPE_VOD)
    m_mediaType = RecordingMediaType::PVR_RECORDING_MEDIA_TYPE_VOD;
else
{
    Clog::Log(..., "Unhandled enum value, defaulting to PVR_RECORDING_MEDIA_TYPE_RECORDING");
    m_mediaType = RecordingMediaType::PVR_RECORDING_MEDIA_TYPE_RECORDING;
}
```

Reply:
Great idea. Added, I also added it to the fill data call.